uber/automated_emails.py

```diff
@@ -96,19 +141,72 @@ def __init__(self, conf):
         AutomatedEmail.__init__(self, Attendee,
                                 subject='{EVENT_NAME} Department Checklist: ' + conf.name,
                                 template='shifts/dept_checklist.txt',
-                                filter=lambda a: a.is_single_dept_head and a.admin_account and days_before(7, conf.deadline) and not conf.completed(a),
+                                filter=lambda a: a.is_single_dept_head and a.admin_account and not conf.completed(a),
+                                date_filters=days_before(7, conf.deadline),
                                 sender=c.STAFF_EMAIL,
                                 extra_data={'conf': conf})
 
-before = lambda dt: bool(dt) and localized_now() < dt
-after = lambda dt: bool(dt) and localized_now() > dt
-days_after = lambda days, dt: bool(dt) and (localized_now() > dt + timedelta(days=days))
 
+print_dateformat = "%m/%d"
+
+
+class days_before:
+    def __init__(self, days, dt, until=None):
+        self.dt = dt
+        self.days = days
+        self.until = until
+
+        if dt:
+            self.starting_date = self.dt - timedelta(days=self.days)
+            self.ending_date = (dt - timedelta(days=until)) if until else dt
+
+    def __call__(self):
+        return self.starting_date < localized_now() < self.ending_date if self.dt else False
+
+    @property
+    def active_when(self):
+        return 'between {} and {}'.format(self.starting_date.strftime(print_dateformat),
+                                          self.ending_date.strftime(print_dateformat)) \
+            if self.dt else ''
+
+
+class days_after:
+    def __init__(self, days, dt):
+        self.dt = dt
+        self.days = days
+
+        self.starting_date = dt + timedelta(days=days) if dt else None
+
+    def __call__(self):
+        return bool(self.dt) and (localized_now() > self.starting_date)
+
+    @property
+    def active_when(self):
+        return 'after {}'.format(self.starting_date.strftime(print_dateformat)) if self.starting_date else ''
```

Comment:
By doing these properties this way, you're encoding styling into an obscure (from a front-end perspective) backend file. We should probably just return the datetime object for this property and let the frontend/views handle what it actually says (don't forget Django's built-in |datetime filter). This also makes this property much more versatile, if we want to use it elsewhere in the backend later on.

By the same token, we may want to name these properties something unique like `active_between` above, `active_after` here and `active_before` below. Alternatively, we can just say something generic like "activation_date" and depend on the class for context (e.g., `days_after.activation_date`)